shipyard-controller/handler/sequencedispatcher.go

```diff
@@ -63,6 +64,12 @@ func (sd *SequenceDispatcher) Add(queueItem models.QueueItem) error {
 			if err2 := sd.sequenceQueue.QueueSequence(queueItem); err2 != nil {
 				return err2
 			}
+		} else if err == ErrSequenceBlockedWaiting {
```

Comment:
I believe we can simplify this as follows:

```
if err := sd.dispatchSequence(queueItem); err != nil {
		if errors.Is(err,ErrSequenceBlocked) || errors.Is(ErrSequenceBlockedWaiting) {
			// if the sequence is currently blocked, insert it into the queue
			if err2 := sd.sequenceQueue.QueueSequence(queueItem); err2 != nil {
				return err2
			}
		} else {
			return err
		}
	}
```

It should not make a difference if the sequence is currently blocked because it is paused, or if another sequence has to be finished first - what we put into the queue should always just be the .triggered event that triggered the sequence, so we don't need to change the event type with th `triggeredToWaitingQueueItem()` function imho.

Reply:
yes, but when the sequence should be waiting, the ErrSequenceBlockedWaiting is returned to the shipyardcontroller.go, so that is the main difference. This is actually needed to discover that onSequenceWaiting() should be executed from the shipyardcontroller.go

Reply:
ah that's true, so we still need to return the appropriate error. However, i do still think we don't need the `triggeredToWaitingQueueItem()` here

Reply:
removed, thanks :)